File: chapter1/reduce.py
import sys

wordmap = {}
for line in sys.stdin:
    word, cnt = line.strip().split("\t")
    if(not wordmap.has_key(word)):
        wordmap[word] = int(cnt)
    else:
        wordmap[word] += int(cnt)

for word in wordmap.keys():
    print("%s\t%d" % (word, wordmap[word]))

File: chapter1/reduce.pl
#!/usr/bin/env perl
# Map function for hadoop streaming using Perl.

our %wordmap={};
while (<STDIN>) {
    $word=$_;
    if($word ne "") {
        if (exists($wordmap{$word})) {
            $wordmap{$word} += 1;
        } else {
            $wordmap{$word} = 1;
        }
    }
}

for my $word (keys %wordmap) {
    my $cnt = $wordmap{$word};
    print $word . "$cnt\n";
}
